### ISSUE TITLE
Think of an awesome name for this package

### ISSUE THREAD

**Issue by thoughtpolice** [well-typed/binary-serialise-cbor#26]
Before we release this package publicly, I think @dcoutts and I both agree the name is _kind of_ a mouthful. It would be nice if this could have a shorter name and module space before we publicly release it and make everyone mad for breaking it.

This should be considered low priority (since, as stated, the eventual plan is for this to become `binary` itself), but would be nice to think about.

**Comment by ghorn** [comment]
How about `cboring` because it (ideally) just works and there are no surprises?

**Comment by arianvp** [comment]
cyborg?

**Comment by axman6** [comment]
Your even just cborg.

**Comment by dcoutts** [comment]
I quite like cborg :-)

**Comment by bgamari** [comment]
@dcoutts says `binary-serialise`.

**Comment by axman6** [comment]
I thought @dcoutts said
> I quite like cborg :-)

=)

**Comment by bgamari** [comment]
@axman6, fair point.

I'm not sure that we want to rename at this point; the package already has a good number of users and renaming would be a cost on them all for negligible benefit; afterall, none of this names particularly stand out.

**Comment by bgamari** [comment]
I've spoken to @hvr and @thoughtpolice and it sounds like there's general consensus that renaming would be worth the effort. There seems to be consensus around the name `cborg`, with the modules rooted at `Serialise.Cborg` (although personally I prefer `Data.Cborg` to avoid breaking with tradition).

**Comment by axman6** [comment]
Personally, I mainly want to see it renamed because binary-serialise-cbor, while an accurate description, is an awful name. A little marketing may go a long way in speeding up adoption too.

**Comment by bgamari** [comment]
After a great deal of discussion, I think @dcoutts and I have finally found a suitable color for this shed which solves a few other related problems as well.

Ultimately, @dcoutts' reluctance over the `cborg` name stemmed from his fear that we would repeat the mistake made in the `binary` library in conflating the concerns of value serialisation and manipulation of arbitrary binary protocols. We have concluded that it would be best to split this package up into two distinct libraries to address this concern:
 * `cborg` (rooted at `Codec.CBOR`) will serve as the low-level substrate, providing support for working with arbitrary CBOR formats. Most of the library will remain here.
 * `serialise` (rooted at `Codec.Serialise`) will serve as the high-level serialisation library. The `Class` module will live here, in addition to re-exports of some subset of `cborg`.
This also allows us to address #120, since the instances will be provided by `serialise`. I'll try to make this happen tonight.
